xlators/protocol/client/src/client-handshake.c

```diff
@@ -808,9 +808,8 @@ client_setvolume_cbk(struct rpc_req *req, struct iovec *iov, int count,
                 op_ret = -1;
                 goto out;
             }
-        } else {
-            strncpy(ctx->volume_id, volume_id, GF_UUID_BUF_SIZE);
-        }
+        } else
+            snprintf(ctx->volume_id, GF_UUID_BUF_SIZE, "%s", volume_id);
```

Comment:
Why is this better than `strncpy()` ?

Reply:
Well, snprintf() should always leave the destination null-terminated. IIUC that's why GCC developers discourages from using strncpy() in general, see https://developers.redhat.com/blog/2018/05/24/detecting-string-truncation-with-gcc-8.